### PR TITLE
Fix ROIPooling op

### PR DESCRIPTION
**Pull request by XingHongChenIntel**
This PR need to work with PR(https://github.com/plaidml/openvino/pull/134) and PR(https://github.com/plaidml/plaidml/pull/1687)

The update include :

**`max` mode:**

- uses 2d tensor instead of 4d tensor to accelerate mode `max`, running time reduced about 30%
- add more boundary check.

**`bilinear` mode:**

- fix boundary error(`roi_h_scale`, `roi_w_scale`).

**test cases:**

- removed a few large tests.
- add `InputLayerType` for test.
- add `smoke` instantiate test.

### CODE REVIEW

File: plaidml/bridge/openvino/tests/functional/shared_tests_instances/single_layer_tests/roi_pooling.cpp [hunk 1]
```diff
@@ -30,23 +29,27 @@ const std::vector<std::vector<size_t>> pooledShapes_bilinear = {
 const std::vector<std::vector<size_t>> coordShapes = {
     {1, 5},
     {3, 5},
-    {5, 5},
 };
 
 const std::vector<InferenceEngine::Precision> netPRCs = {
-    InferenceEngine::Precision::FP16,
```
Comment (tzerrell) [inline plaidml/bridge/openvino/tests/functional/shared_tests_instances/single_layer_tests/roi_pooling.cpp:37]: Are the FP16 tests broken or are you just trying to reduce the testing size? Knowing whether FP16 is working is important, and if FP16 is working we should retain the FP16 tests even if they are slow. (If FP16 is broken we can omit FP16 tests like you're doing here, but we will need to track the work of eventually fixing them.)

**Review by tzerrell — COMMENTED** [review]
Looks generally good; I have some inline comments

File: plaidml/bridge/openvino/ops/roi_pooling.cpp [hunk 2]
```diff
@@ -2,88 +2,101 @@
...
+  float roi_width = std::max(roi_w_end - roi_w_start + 1, 1.0f);
+  float roi_height = std::max(roi_h_end - roi_h_start + 1, 1.0f);
 
-  auto bin_size_h = roi_height / static_cast<float>(pooled_h);
-  auto bin_size_w = roi_width / static_cast<float>(pooled_w);
+  float bin_size_h = roi_height / pooled_h;
+  float bin_size_w = roi_width / pooled_w;
 
-  edsl::Tensor pooled_tensor;
+  // edsl::Tensor pooled_tensor;
```
[the remark this answers is not in the document]
Comment (XingHongChenIntel) [reply plaidml/bridge/openvino/ops/roi_pooling.cpp]: removed this comment.

File: plaidml/bridge/openvino/tests/functional/shared_tests_instances/single_layer_tests/roi_pooling.cpp (hunk 1 above)
Comment (XingHongChenIntel) [reply plaidml/bridge/openvino/tests/functional/shared_tests_instances/single_layer_tests/roi_pooling.cpp:37]: yes, FP16 tests are broken with `bilinear` mode. for now, I just comment it, but we find that FP16 issue shows in many other ops, we thought it maybe a common issue, and will fix it in a separate PR.

Comment (tzerrell) [reply plaidml/bridge/openvino/tests/functional/shared_tests_instances/single_layer_tests/roi_pooling.cpp:37]: Yes, makes sense.

**Review by tzerrell — APPROVED** [review]

**Review by tzerrell — APPROVED** [review]

**Pull request merged**